main/options/options.cc

```diff
@@ -956,6 +975,7 @@ void readOptions(Options &opts,
             throw EarlyReturnWithCode(1);
         }
         opts.stopAfterPhase = extractStopAfter(raw, logger);
+        opts.parser = extractParser(raw, logger);
```

Comment:
Given the level of current support for LSP, we are leaning towards disallowing `--parser=prism` when combined with `--lsp`; the necessity of that would depend on the LSP testcase support when this PR is closer to merging.